### PR TITLE
Parameterize test_link_params

### PR DESCRIPTION
**Pull request by areusch**
Cleaning up test_link_params to support the new parameterized testing.

cc @Lunderberg

### CODE REVIEW

File: tests/python/unittest/test_link_params.py [hunk 1]
```diff
@@ -181,54 +183,53 @@ def _add_decl(name, dtype):
...
+    ir_mod, param_init = _make_mod_and_params(linkable_dtype)
+    rand_input = _make_random_tensor(linkable_dtype, INPUT_SHAPE)
+    main_func = ir_mod["main"]
+    target = "llvm --runtime=c --system-lib --link-params"
+    with tvm.transform.PassContext(opt_level=3):
+        lib = tvm.relay.build(ir_mod, target, params=param_init)
+
+        # NOTE: Need to export_library() and load_library() to link all the Module(llvm, ...)
+        # against one another.
+        temp_dir = tempfile.mkdtemp()
```
[the remark this answers is not in the document]
Comment (areusch) [reply tests/python/unittest/test_link_params.py]: fixed this one

File: tests/python/unittest/test_link_params.py [hunk 2]
```diff
@@ -181,54 +183,53 @@ def _add_decl(name, dtype):
...
-                return graph_rt.get_output(0)
-
-            unlinked_output = _run_unlinked(lib)
-
-        if "int" in dtype:
-            np.testing.assert_equal(unlinked_output.numpy(), linked_output.numpy())
-        else:
-            np.testing.assert_allclose(unlinked_output.numpy(), linked_output.numpy())
+def test_llvm_link_params(linkable_dtype):
+    ir_mod, param_init = _make_mod_and_params(linkable_dtype)
```
[the remark this answers is not in the document]
Comment (areusch) [reply tests/python/unittest/test_link_params.py:186]: i'd like to address this one, but i'm a little short on time right now. hopefully we can merge this in the spirit of incremental progress and address later

**Comment by areusch** [comment]
@Lunderberg I sync'd this one up, PTAL when you get a chance.

File: tests/python/unittest/test_link_params.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lunderberg) [reply tests/python/unittest/test_link_params.py:186]: Sounds good, and agreed.

**Pull request merged**